src/inmanta/parser/__init__.py

```diff
@@ -58,3 +58,26 @@ class SyntaxDeprecationWarning(ParserWarning):
 
     def __init__(self, location: Range, value: object, msg: str) -> None:
         ParserWarning.__init__(self, location, value, msg)
+
+
+class InvalidNamespaceAccess(ParserException):
+    """
+    Exception raised when a namespace access is attempted with '.' rather than '::'.
+    """
+
+    def __init__(self, invalid: LocatableString) -> None:
+        self.invalid: LocatableString = invalid
+        super().__init__(
+            location=invalid.location,
+            value=str(invalid),
+            msg=(
+                f"invalid namespace access `{invalid}`. Namespaces should be accessed with '::' rather than '.'"
+                f" (reserved for attribute and relation access): `{self.suggest_replacement()}`"
+            ),
```

Comment:
I don't think it's sufficiently clear on which of both separators the text between brackets reflects.
```suggestion
                f"invalid namespace access `{invalid}`. Namespaces should be accessed with '::' rather than '.'. "
                f"The '.' separator is reserved for attribute and relation access. Did you mean: `{self.suggest_replacement()}`"
            ),
```